Cat2Cat/main.m

```diff
@@ -207,10 +218,34 @@ int main(int argc, const char * argv[])
         [assetCatalogPaths addObjectsFromArray:filesMatchingPattern(url.path)];
     }
     parameters.assetCatalogPaths = assetCatalogPaths;
-    
+
+    parameters.methodNamePrefix = methodNamePrefix;
+
     NSURL *outputUrl = [NSURL URLWithString:[outputDirectory stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]
                               relativeToURL:basePathUrl];
     parameters.outputDirectory = outputUrl.path;
     
     return parameters;
 }
+
+static BOOL validateMethodNamePrefix(NSString *methodNamePrefix, NSError **error)
+{
+    if ([methodNamePrefix length] < 2) {
+        *error = [NSError errorWithDomain:@"Cat2Cat" code:-1 userInfo:@{ NSLocalizedDescriptionKey : @"Method name prefix must be at least 2 characters." }];
+    } else {
+        NSMutableCharacterSet *charSet = [[NSCharacterSet letterCharacterSet] mutableCopy];
+        [charSet addCharactersInString:@"_"];
+
+        if ([methodNamePrefix rangeOfCharacterFromSet:[charSet invertedSet]].location != NSNotFound) {
+            *error = [NSError errorWithDomain:@"Cat2Cat" code:-1 userInfo:@{ NSLocalizedDescriptionKey : @"Method name prefix contains invalid characters." }];
+        }
+    }
+    return *error == nil;
```

Comment:
My inclination is that it's up to the caller to decide whether or not they care about the error information and that the function itself should be as defensive as possible.  @designatednerd ?